File: 1. Array & Hashing/49. 字母异位词分组/49. 字母异位词分组.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        res = []
        dic = {}
        for i, sub_s in enumerate(strs):
            count = [0]*26
            for s in sub_s:
                count[ord(s)-97] += 1
            if tuple(count) not in dic:
                dic[tuple(count)] = []
            dic[tuple(count)].append(sub_s)
        for k in dic.values():
            res.append(k)
        return res

File: Array & Hashing/49. 字母异位词分组/49. 字母异位词分组.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> res;
        unordered_map<string , vector<string>> map;
        for(int i = 0; i < strs.size(); i++){
            string k = getkey(strs[i]);
            map[k].push_back(strs[i]);
        }

        for (auto it = map.begin(); it != map.end(); it++) {
            res.push_back((*it).second);
        }        
    return res;
    }

    string getkey(string str){
        vector<int> temp(26);
        string key = "";
        for(int i = 0; i < str.size(); i++){
            //char與char以10進位相減
            temp[str[i] - 'a']++;
        }
        //依序排列
        for(int i = 0; i < temp.size(); i++){
            key += to_string(temp[i] + 'a');
        }
        return key;
    }
};